tinyagentos/notifications.py

```
@@ -217,80 +242,155 @@ async def list(
-            conds.append("(user_id IS NULL OR user_id = ?)")
-        params: tuple = (user_id,) if user_id is not None else ()
-        async with self._db.execute(
-            f"SELECT COUNT(*) FROM notifications WHERE {' AND '.join(conds)}",
-            params,
-        ) as cursor:
+            # For per-user notifications, include both own notifications and broadcasts
+            # Read state for broadcasts is stored in notification_user_state
+            sql = (
+                "SELECT COUNT(*) FROM notifications "
+                "LEFT JOIN notification_user_state nus ON n.id = nus.notification_id AND nus.user_id = ? "
+                f"WHERE (n.user_id IS NULL OR n.user_id = ?) AND archived = 0 AND COALESCE(nus.read_at, n.read) = 0"
```

Ruff (0.16.3)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)